crates/cdk/src/wallet/keysets.rs

```diff
@@ -86,6 +86,21 @@ impl Wallet {
             }
         }
 
-        active_keysets.first().ok_or(Error::NoActiveKeyset).cloned()
+        Ok(active_keysets)
+    }
+
+    /// Get active keyset for mint with the lowest fees
+    ///
+    /// Queries mint for current keysets then gets [`Keys`] for any unknown
+    /// keysets
+    #[instrument(skip(self))]
+    pub async fn get_active_mint_keyset(&self) -> Result<KeySetInfo, Error> {
+        let active_keysets = self.get_active_mint_keysets().await?;
+
+        let keyset_with_lowest_fee = active_keysets
+            .into_iter()
+            .min_by_key(|key| key.input_fee_ppk)
+            .ok_or(Error::NoActiveKeysetWithLowestFee)?;
```

Comment:
```suggestion
            .ok_or(Error::NoActiveKeyset)?;
```

This will only return a `None` if there is no active keyset so we don't need a new error variant we can just use `NoActiveKeyset`